pluma/pluma-view.c

```diff
@@ -794,7 +794,7 @@ pluma_override_font (const gchar          *item,
 		     GtkWidget            *widget,
 		     PangoFontDescription *font)
 {
-	static GtkCssProvider *provider = NULL; /*We need to keep this as long as Pluma is running*/
+	GtkCssProvider *provider = NULL;
```

Comment:
This change makes these cppcheck warnings:

```
pluma/pluma-view.c:820:15: style: Condition 'provider==NULL' is always true [knownConditionTrueFalse]
 if (provider == NULL)
              ^

pluma/pluma-view.c:797:29: note: Assignment 'provider=NULL', assigned value is 0
 GtkCssProvider *provider = NULL;
                            ^

pluma/pluma-view.c:820:15: note: Condition 'provider==NULL' is always true
 if (provider == NULL)
              ^
```

provider must to be static to make it only one time, if we do the same more times we have memory leaks

Reply:
the problem is `gtk_style_context_add_provider_for_screen` sets style for the minimap as well, such that its font becomes much too large. Also, `GdkScreen` is removed in Gtk4.
Does it make sense to just remove the if statement and add `g_clear_object (&provider)` to the end?

Reply:
my review is just in this line, did you test it?

I have tested and `"static GtkCssProvider *provider = NULL;"` works here, the minimap works

Reply:
`GtkCssProvider *provider` should be a static variable. `pluma_override_font` is called for overriding `textview` and `label` fonts, for this reason `GtkCssProvider *provider` is a global variable.

Reply:
Overwriting the global (for screen) provider only for the minimap might be a better idea then?

Leaving PlumaView untouched and adding:
```diff
	tab->priv->view_map_frame = gtk_frame_new (NULL);
	map = gtk_source_map_new();

+	GtkCssProvider *provider = gtk_css_provider_new ();
+	gtk_css_provider_load_from_data (provider,
+					 "textview { font-family: Monospace; font-size: 1pt; }",
+					 -1,
+					 NULL);
+	gtk_style_context_add_provider (gtk_widget_get_style_context (map),
+					GTK_STYLE_PROVIDER (provider),
+					GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+	g_object_unref (provider);

	gtk_source_map_set_view (GTK_SOURCE_MAP(map), GTK_SOURCE_VIEW(tab->priv->view));
	gtk_container_add (GTK_CONTAINER(tab->priv->view_map_frame), map);
	gtk_widget_show (tab->priv->view_map_frame);
```
seems to work for me.

Reply:
No, GtkSourceMap already has monospace in Style Classes.

### GTK Inspector

- PlumaView

![Screenshot at 2020-05-21 11-13-41](https://user-images.githubusercontent.com/10171411/82544062-d321fe80-9b54-11ea-9b72-aef029e3c0fa.png)

- GtkSourceMap

![Screenshot at 2020-05-21 11-12-42](https://user-images.githubusercontent.com/10171411/82544075-d87f4900-9b54-11ea-8fd5-6dbba80f0ee9.png)

Reply:
Not if it gets overwritten by `gtk_style_context_add_provider_for_screen` in PlumaView's `pluma_override_font`:
![Screenshot at 2020-05-21 12-53-36](https://user-images.githubusercontent.com/39454100/82552431-5e55c100-9b62-11ea-8148-4863c14ac54b.png)

Reply:
I think we got different results because we are reviewing different things, please update your PR and we'll review if theming and font overwriting works properly.
```patch
diff --git a/pluma/pluma-view.c b/pluma/pluma-view.c
index 5280194..f2f9e02 100644
--- a/pluma/pluma-view.c
+++ b/pluma/pluma-view.c
@@ -794,7 +794,7 @@ pluma_override_font (const gchar          *item,
                     GtkWidget            *widget,
                     PangoFontDescription *font)
 {
-       GtkCssProvider *provider = NULL;
+       static GtkCssProvider *provider = NULL;
        gchar          *prov_str;
        gchar          *css;
        gchar          *family;
@@ -825,9 +825,9 @@ pluma_override_font (const gchar          *item,
                                  "changed::" "font-name",
                                  G_CALLBACK (system_font_changed_cb), NULL);
 
-               gtk_style_context_add_provider (gtk_widget_get_style_context (widget),
-                                               GTK_STYLE_PROVIDER (provider),
-                                               GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+               gtk_style_context_add_provider_for_screen (gtk_widget_get_screen (widget),
+                                                          GTK_STYLE_PROVIDER (provider),
+                                                          GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
        }
 
        prov_str = gtk_css_provider_to_string (provider);
```